### PR TITLE
[PB-11] functionality to edit genre button implemented

### PR DESCRIPTION
**Pull request by Paulina594**

### CODE REVIEW

File: src/app/components/genre-form-dialog/genre-form-dialog.component.html [hunk 1]
```diff
@@ -1,12 +1,12 @@
-<h1 mat-dialog-title>Add new genre</h1>
+<h1 mat-dialog-title>{{modalTitle}}</h1>
 <div mat-dialog-content [formGroup]="form">
     <mat-form-field appearance="fill">
-      <mat-label>New genre</mat-label>
+      <mat-label>{{placeholder}}</mat-label>
       <input matInput formControlName="name" >
-      <mat-error *ngIf="form.get('name').invalid">Please, provide genre name!</mat-error>
+      <mat-error *ngIf="form.get('name').errors?.required">This field can't be empty!</mat-error>
```
Comment (sdrozdz) [inline src/app/components/genre-form-dialog/genre-form-dialog.component.html]: Add error for `uniqueName`

File: src/app/components/genre-form-dialog/genre-form-dialog.component.ts [hunk 2]
```diff
@@ -15,18 +25,47 @@ import {
...
+      this.placeholder = 'New genre';
+      this.name = '';
+      this.editMode = false;
+    }
+
     this.form = this.fb.group({
-      name: new FormControl('', Validators.required),
+      name: new FormControl(this.name, [
+        Validators.required,
+        Validators.pattern('([a-zA-Z ])*'),
```
Comment (sdrozdz) [inline src/app/components/genre-form-dialog/genre-form-dialog.component.ts:56]: Add error message for pattern validator

File: src/app/components/genre-form-dialog/genre-form-dialog.component.ts [hunk 3]
```diff
@@ -37,9 +76,32 @@ export class GenreFormDialogComponent implements OnInit {
...
+      ...this.form.value,
+      name: capitalizeValue,
+    };
+
+    const saveMethod: Observable<Genre> = this.data.editMode
+      ? this.genreService.editGenre(this.data.genre.id, value.name)
+      : this.genreService.saveGenre(value.name);
+
+    saveMethod.subscribe((responseData) => {
+      console.log(responseData);
```
Comment (sdrozdz) [inline src/app/components/genre-form-dialog/genre-form-dialog.component.ts]: remove console log

File: src/app/components/genre-form-dialog/genre-form-dialog.component.ts [hunk 4]
```diff
@@ -15,18 +25,47 @@ import {
...
+      this.modalTitle = 'Add new genre';
+      this.submitTitle = 'Save';
+      this.placeholder = 'New genre';
+      this.name = '';
+      this.editMode = false;
+    }
+
     this.form = this.fb.group({
-      name: new FormControl('', Validators.required),
+      name: new FormControl(this.name, [
```
Comment (sdrozdz) [inline src/app/components/genre-form-dialog/genre-form-dialog.component.ts]: Replace `this.name` with `this.data.name ?: ''` and then remove name field from class

**Review by sdrozdz — CHANGES_REQUESTED** [review]
Good overall but some changes are needed:

File: src/app/components/genre-form-dialog/genre-form-dialog.component.html (hunk 1 above)
Comment (Paulina594) [reply src/app/components/genre-form-dialog/genre-form-dialog.component.html]: added

File: src/app/components/genre-form-dialog/genre-form-dialog.component.ts (hunk 2 above)
Comment (Paulina594) [reply src/app/components/genre-form-dialog/genre-form-dialog.component.ts:56]: added

File: src/app/components/genre-form-dialog/genre-form-dialog.component.ts (hunk 3 above)
Comment (Paulina594) [reply src/app/components/genre-form-dialog/genre-form-dialog.component.ts]: removed

File: src/app/components/genre-form-dialog/genre-form-dialog.component.ts (hunk 4 above)
Comment (Paulina594) [reply src/app/components/genre-form-dialog/genre-form-dialog.component.ts]: updated

**Review by sdrozdz — APPROVED** [review]
approved-target-lady

**Pull request merged**